core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -708,7 +711,8 @@ private void failReplicaIfNeeded(String index, int shardId, Throwable t) {
          */
         public ReplicationPhase(ShardIterator originalShardIt, ReplicaRequest replicaRequest, Response finalResponse,
                                 ClusterStateObserver observer, ShardRouting originalPrimaryShard,
-                                InternalRequest internalRequest, ActionListener<Response> listener, Releasable indexShardReference) {
+                                InternalRequest internalRequest, ActionListener<Response> listener, Releasable indexShardReference,
+                                ShardStateActionListener shardStateListener) {
```

Comment:
I've changed the tests in 587c67f9f1e8b48b68df686af68f9bf3a5d3490c and removed the field in acb54ab981be65cf8a1ac2f9d884d369b908ebd2.